src/entity-schema.js

```diff
@@ -28,19 +32,19 @@ export function entitySchema(entity) {
       type: types[key]
     };
 
-    element = extend(true, element, data[key] ? data[key].form || {} : {});
-
-    if (associations[key]) {
-      element.key      = key;
+    if (isAssociation(key)) {
+      element.type     = 'association';
       element.resource = associations[key].entity;
+      element.property = 'name';
     }
 
-    if (associations[key] && associations[key].type === 'collection') {
-      element.type     = 'association';
+    if (isAssociation(key) && isCollectionAssociation(key)) {
       element.multiple = true;
       element.schema   = entitySchema(entityManager.getEntity(key));
     }
 
+    element = extend(true, element, data[key] ? data[key].form || {} : {});
```

Comment:
extend accepts null and undefined afaik.
 so one could write just (data[key] || {}).form
not that this would be more readable :smile: